src/darwin/list.mm

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2017 resin.io
+ * Copyright 2018 Robin Andersson <me@robinwassen.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <nan.h>
+#include "../drivelist.hpp"
+
+#import "REDiskList.h"
+#import <Cocoa/Cocoa.h>
+#import <DiskArbitration/DiskArbitration.h>
+
+namespace Drivelist {
+  bool IsDiskPartition(NSString *disk) {
+    NSPredicate *partitionRegEx = [NSPredicate predicateWithFormat:@"SELF MATCHES %@", @"disk\\d+s\\d+"];
+    return [partitionRegEx evaluateWithObject:disk];
+  }
+
+
+  bool IsCard(CFDictionaryRef diskDescription) {
+    CFDictionaryRef mediaIconDict = (CFDictionaryRef)CFDictionaryGetValue(diskDescription, kDADiskDescriptionMediaIconKey);
+    NSString * iconFileName = (NSString*)CFDictionaryGetValue(mediaIconDict,
+                                                              CFStringCreateWithCString(NULL, "IOBundleResourceFile", kCFStringEncodingUTF8));
+
+    return [iconFileName isEqualToString:@"SD.icns"];
+  }
+
+
+  NSNumber *DictNum(CFDictionaryRef dict, const void *key) {
+      return (NSNumber*)CFDictionaryGetValue(dict, key);
```

Comment:
Like here, there are some inconsistencies in indentation – can you normalize to 2 space indentation (since that's what we're using in this repo)?